### PR TITLE
Ported popover overhaul PR from Calypso.

### PR DESCRIPTION
**Pull request by zinigor**
This is almost a total copy/paste of the PR by @retrofox: https://github.com/Automattic/wp-calypso/pull/7217

With it we can solve the ghosting problem, but the styles need some love. Currently stats popovers in Jetpack look like this:

popover

cc @jeffgolenski and @MichaelArestad can you please work your magic either here or in the Jetpack repo? Thank you!

### CODE REVIEW

**Comment by retrofox** [comment]
IMHO we should export these components individually.

```es6
import Popover from 'calypso/popover';
```

... or something like that.

**Comment by zinigor** [comment]
@retrofox if I understand you're opposed to the whole idea of dops-components being a separate repo?

**Comment by retrofox** [comment]
No necessarily I'm opposed to dots-components. I think is completely ok build this pack of components (?).
What I mean that I think doesn't make sense copy and paste the code of a component when we have the chance to publish this component and then import it from whatever we need since `calypso`, `dots-components`, etc.

It's just a comment because I thing we've discussed this approach many times. But if someone is thinking/wondering if we should/could do it I'll be happy to raise my thumb.

**Comment by MichaelArestad** [comment]
Wait. What happened? The last time I looked at stats last week, the popover was grey.

**Comment by zinigor** [comment]
Yes it was :) And now due to the changes in markup I'm betting the styles no longer apply, can you take a look at what's missing please?

File: client/components/chart/bar.jsx [hunk 1]
```diff
@@ -80,13 +81,62 @@ module.exports = React.createClass( {
 
 
 	mouseEnter: function(){
+		console.log( 'Mouse in' );
```
Comment (MichaelArestad) [inline client/components/chart/bar.jsx]: Do we need these?

Comment (zinigor) [inline client/components/chart/bar.jsx]: Uh oh! Good find, I'll remove those.

**Comment by zinigor** [comment]
I have removed debugging output, docs and tests that were added when merging, thanks @MichaelArestad ! Do not merge this yet, though, let's wait for Jetpack 4.3.1 to ship first.

**Comment by MichaelArestad** [comment]
@zinigor Can this be merged now?

**Comment by MichaelArestad** [comment]
Fixes https://github.com/Automattic/jetpack/issues/4945

**Comment by zinigor** [comment]
Yes, let's ship it in 4.3.2

**Pull request merged**